packages/stacking/src/index.ts

```diff
@@ -0,0 +1,410 @@
+   */
+  async hasMinimumRequiredStxAmount(): Promise<boolean> {
+    const balance: BN = await this.getAccountBalance();
+    // TODO pox info should use string type instead of number
+    const min: BN = new BN((await this.getPoxInfo()).min_amount_ustx.toString());
+    return balance.gte(min);
+  }
+
+  /**
+   * Check if account can lock stx
+   * 
+   * @param {CanLockStxOptions} options - a required lock STX options object 
+   * 
+   * @returns {Promise<StackingEligibility>} that resolves to a StackingEligibility object if the operation succeeds
+   */
+  async canLockStx({
+    poxAddress, 
+    cycles
+  }: CanLockStxOptions
+  ): Promise<StackingEligibility> {
+    const balancePromise: Promise<BN> = this.getAccountBalance();
+    const poxInfoPromise = this.getPoxInfo();
+
+    return Promise.all([balancePromise, poxInfoPromise])
+      .then(([balance, poxInfo]) => {
+        const { hashMode, data } = convertBTCAddress(poxAddress);
+        const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+        const hashbytes = bufferCV(data);
+        const poxAddressCV = tupleCV({
+          hashbytes,
+          version: hashModeBuffer,
+        });
+
+        const [contractAddress, contractName] = poxInfo.contract_id.split('.');
+
+        return callReadOnlyFunction({
+          network: this.network,
+          contractName,
+          contractAddress,
+          functionName: 'can-stack-stx',
+          senderAddress: this.address,
+          functionArgs: [
+              poxAddressCV,
+              uintCV(balance.toString()),
+              uintCV(poxInfo.reward_cycle_id),
+              uintCV(cycles.toString())
+            ],
+        });
+      })
+      .then((responseCV: ClarityValue) => {
+        if(responseCV.type === ClarityType.ResponseOk) {
+          return {
+            eligible: true
+          };
+        } else {
+          const errorCV = responseCV as ResponseErrorCV;
+          return {
+            eligible: false,
+            reason: StackingErrors[+cvToString(errorCV.value)]
+          };
+        }
+      })
+  }
+
+  /**
+   * Generate and broadcast a stacking transaction to lock STX
+   * 
+   * @param {LockStxOptions} options - a required lock STX options object 
+   * 
+   * @returns {Promise<string>} that resolves to a broadcasted txid if the operation succeeds
+   */
+  async lockStx({
+    amountMicroStx,
+    poxAddress,
+    cycles,
+    key,
+    burnBlockHeight,
+  }: LockStxOptions
+  ) {
+    const poxInfo = await this.getPoxInfo();
+    const contract = poxInfo.contract_id;
+
+    const txOptions = this.getLockTxOptions({
+      amountMicroStx,
+      cycles,
+      poxAddress,
+      contract,
+      burnBlockHeight,
+    });
+    const tx = await makeContractCall({
+      ...txOptions,
+      senderKey: key,
+    });
+
+    const res = await broadcastTransaction(tx, txOptions.network as StacksNetwork);
+    if (typeof res === 'string') {
+      return res;
```

Comment:
shouldn't this also return the explorer URL?

Reply:
I think the consumer of the lib should construct the link vs in the lib

Reply:
why? I think we should do it in order to avoid the dev building the URLs - they'd want to use the same URL that we would provide anyways, right? One reason I see for not doing this is if we were to stay open to other explorers and not prioritizing one over the other, but that doesn't seem relevant right now

Reply:
We should maintain separation of concern in the code. The URL can be easily constructed by the developer should they need it. And we shouldn't favour any explorer over another, so we shouldn't start doing that here. 

I don't think the goal should be to allow the developer to write as few lines of code as possible. And we should keep in mind good engineering practices and maintainability of code.

Reply:
I agree, I think it's not the responsibility of this package to push people to the explorer, but whatever would be using this package to expose the url

Reply:
as per discussion in the meeting, we can close this out